tests/test_fermionic_solver.py

```diff
@@ -0,0 +1,275 @@
+
+
+    #Pade cut-off
+    lmax =10
+
+
+    w_list = np.linspace(-2,2,100)
+    def deltafun(j,k):
+        if j==k: 
+            return 1.
+        else:
+            return 0.
+    def Gamma_L_w(w):
+        return Gamma*W**2/((w-mu_l)**2 + W**2)
+
+    def Gamma_w(w, mu):
+        return Gamma*W**2/((w-mu)**2 + W**2)
+
+
+    def f(x):
+        kB=1.
+        return 1/(np.exp(x)+1.)
+
+
+    Alpha =np.zeros((2*lmax,2*lmax))
+    for j in range(2*lmax):
+        for k in range(2*lmax):
+            Alpha[j][k] = (deltafun(j,k+1)+deltafun(j,k-1))/sqrt((2*(j+1)-1)*(2*(k+1)-1))
+            
+    eigvalsA=eigvalsh(Alpha)  
+
+    eps = []
+    for val in  eigvalsA[0:lmax]:
+        #print(-2/val)
+        eps.append(-2/val)
+        
+    AlphaP =np.zeros((2*lmax-1,2*lmax-1))
+    for j in range(2*lmax-1):
+        for k in range(2*lmax-1):
+            AlphaP[j][k] = (deltafun(j,k+1)+deltafun(j,k-1))/sqrt((2*(j+1)+1)*(2*(k+1)+1))
+            #AlphaP[j][k] = (deltafun(j,k+1)+deltafun(j,k-1))/sqrt((2*(j+2)-1)*(2*(k+2)-1))
+            
+    eigvalsAP=eigvalsh(AlphaP)    
+
+    chi = []
+    for val in  eigvalsAP[0:lmax-1]:
+        #print(-2/val)
+        chi.append(-2/val)
+
+        
+    eta_list = [0.5*lmax*(2*(lmax + 1) - 1)*( 
+      np.prod([chi[k]**2 - eps[j]**2 for k in range(lmax - 1)])/
+        np.prod([eps[k]**2 - eps[j]**2 +deltafun(j,k) for k in range(lmax)])) 
+              for j in range(lmax)]
+
+
+
+    kappa = [0]+eta_list
+
+    epsilon = [0]+eps
+ 
+
+
+    def f_approx(x):
+        f = 0.5
+        for l in range(1,lmax+1):
+            f= f - 2*kappa[l]*x/(x**2+epsilon[l]**2)
+        return f
+
+    def C(sigma,mu):
+        eta_list = []
+        gamma_list  =[]
+        
+
+        eta_0 = 0.5*Gamma*W*f_approx(1.0j*beta*W)
+
+        gamma_0 = W - sigma*1.0j*mu
+        eta_list.append(eta_0)
+        gamma_list.append(gamma_0)
+        if lmax>0:
+            for l in range(1,lmax+1):
+                eta_list.append(-1.0j*(kappa[l]/beta)*Gamma*W**2/(-(epsilon[l]**2/beta**2)+W**2))
+                gamma_list.append(epsilon[l]/beta - sigma*1.0j*mu)
+        
+        return eta_list, gamma_list
+
+   
+   
+   
+    
+
+
+    etapL,gampL = C(1.0,mu_l)
+
+    etamL,gammL = C(-1.0,mu_l)
+
+
+    etapR,gampR = C(1.0,mu_r)
+    etamR,gammR = C(-1.0,mu_r)
+    
+    #heom simulation with above params (Pade)
+    options = Options(nsteps=15000, store_states=True, rtol=1e-14, atol=1e-14)
+
+    #Single fermion.
+    d1 = destroy(2)
+
+    #Site energy
+    e1 = 1. 
+
+
+    H0 = e1*d1.dag()*d1 
+
+    #There are two leads, but we seperate the interaction into two terms, labelled with \sigma=\pm
+    #such that there are 4 interaction operators (See paper)
+    Qops = [d1.dag(),d1,d1.dag(),d1]
+
+
+
+    Kk=lmax+1
+    Ncc = 2  #For a single impurity we converge with Ncc = 2
+    
+    eta_list = [etapR,etamR,etapL,etamL]
+    gamma_list = [gampR,gammR,gampL,gammL]
+    Qops = [d1.dag(),d1,d1.dag(),d1]
```

Comment:
This seems to overwrite the previous definition of `Qops` with the same value it had?

Reply:
just making sure it sticks :)